src/helpers/parse_snak.ts

```diff
@@ -105,20 +108,19 @@ for (const [ datatype, parser ] of Object.entries(parsers)) {
   normalizedParsers[normalizeDatatype(datatype)] = parser
 }
 
-export function parseClaim (datatype, datavalue, options, claimId) {
-  // Known case of missing datatype: form.claims, sense.claims, mediainfo.statements
+export function parseSnak (datatype: DataType | void, datavalue: SnakValue, options: SimplifySnakOptions) {
+  // @ts-expect-error Known case of missing datatype: form.claims, sense.claims, mediainfo.statements
   datatype = datatype || datavalue.type
 
   try {
     // Known case requiring normalization
-    // - legacy "muscial notation" datatype
+    // - legacy "musical notation" datatype
     // - mediainfo won't have datatype="globe-coordinate", but datavalue.type="globecoordinate"
     const parser = normalizedParsers[normalizeDatatype(datatype)]
     return parser(datavalue, options)
   } catch (err) {
     if (err.message === 'parsers[datatype] is not a function') {
       err.message = `${datatype} claim parser isn't implemented
-      Claim id: ${claimId}
       Please report to https://github.com/maxlath/wikibase-sdk/issues`
     }
     throw err
```

Comment:
this try/catch could be simplified by an if. That would be also way better performance wise and will work with more strict typescript checks (err is unknown and needs to be checked, parser can be undefined as it is an array access).

```ts
// Known case requiring normalization
// - legacy "muscial notation" datatype
// - mediainfo won't have datatype="globe-coordinate", but datavalue.type="globecoordinate"
const parser = normalizedParsers[normalizeDatatype(datatype)]
if (!parser) {
  throw new Error(`${datatype} claim parser isn't implemented\nPlease report to https://github.com/maxlath/wikibase-sdk/issues`)
}

return parser(datavalue, options)
```

Reply:
done in d77cbd8